File: Chapter01/binarySearch.py
# 二分查找
from typing import List

def binarySearch(list:list[int],item:int):
    """[二分查找有序列表index]

    Args:
        list (list[int]): [有序列表]
        item (int): [目标值]

    Returns:
        [type]: [int|None]
    """
    low=0
    high=len(list)-1

    while(low<=high):
        mid=(low+high)//2
        guess=list[mid]
        if(guess==item):
            return mid
        if(guess>item):
            high=mid-1
        else:
            low=mid+1
    return None

my_list=[1,3,5,7,9]
print(binarySearch(my_list,3))
print(binarySearch(my_list,-1))

File: Chapter01/binarySearch.js
/* 二分查找 */
/**
 * @description 二分查找
 * @param {Array} list 有序列表
 * @param {Number} desValue 目标值
 * @returns 
 */
function binarySearch(list,desValue){
    let low=0;
    let high=list.length-1;
    while(low<=high){
        let mid=Math.floor((low+high)/2);
        let guess=list[mid];
        if(guess===desValue) return mid;
        if(desValue>guess){
            low=mid+1;
        }else{
            high=mid-1;
        }
    }
    return undefined;
}

let my_list=[1,3,5,7,9];
console.log(binarySearch(my_list,7));